Package.swift

```diff
@@ -41,14 +47,18 @@ let package = Package(
         .library(name: "GRDB", targets: ["GRDB"]),
         .library(name: "GRDB-dynamic", type: .dynamic, targets: ["GRDB"]),
     ],
+    traits: [
+        .trait(name: "GRDBCIPHER", description: "Use the SQLCipher library rather than the vendored SQLite"),
```

Comment:
After reading the naming concerns expressed in https://github.com/groue/GRDB.swift/issues/1816, and the trait names in [SE-0450](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0450-swiftpm-package-traits.md), I suggest using the name `SkipSQLCipher`, and a description that mentions the repository URL https://github.com/skiptools/swift-sqlcipher

Reply:
We're about to have multiple SQLCipher versions: the one in this pull request, the one from #1816, and the CocoaPods one (even if it's high time I completely removed it).

To simplify the compiler flags dance, I start working on replacing most uses of `#if GRDBCIPHER` with `#if SQLITE_HAS_CODEC`. That last flag ought to be set by all GRDB+SQLCipher flavors.

cc @R4N